game_event/tests.py

```diff
@@ -31,6 +34,34 @@ def saved_game_event_player(saved_game_event, player):
                                           ball_responsible=False)
 
 
+@pytest.fixture
+def court_ball_game(court):
+    return CourtBallGame.objects.create(court=court, ball_game=TEST_BALL_GAME)
+
+
+@pytest.fixture
+def mock_urlopen():
+    with mock.patch('urllib.request.urlopen') as mock_urlopen:
+        return mock_urlopen
```

Comment:
Patching is the act of replacing something within another object by a mock. In this case, you just want to return the mock, so you don't need to patch.
```suggestion
    return mock.Mock()
```

Creating the mock this way, means that it will accept anything you'd throw its way. e.g. it can run `mock_urlopen.does_not_exist`

You can also create more accurate mocks, which is encouraged, which will only pretend to have the capabilities of the mocked object. You do this with `create_autospec`:
```suggestion
    return mock.create_autospec(urllib.request.urlopen)
```

See the difference here:
```
In [9]: gen_mock = mock.Mock()

In [10]: gen_mock.say_ahh()
Out[10]: <Mock name='mock.say_ahh()' id='140540985628320'>

In [11]: spec_mock = mock.create_autospec(urllib.request.urlopen)

In [12]: spec_mock.say_ahh()
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In [12], line 1
----> 1 spec_mock.say_ahh()

AttributeError: 'function' object has no attribute 'say_ahh'

In [13]: spec_mock('some-url')
Out[13]: <MagicMock name='mock()' id='140541262404672'>
```

All in all:
- You don't need the patch and the context manager, just return the mock.
- use autospec instead of generic mocks when possible.

Reply:
@yftacherzog Went with your advice. Fixed.